2_eda.ipynb

```diff
@@ -5285,86 +5891,110 @@
        "\n",
        "      cons.price.idx  cons.conf.idx  euribor3m  nr.employed  y  \\\n",
-       "26387         93.200          -42.0      4.076       5195.8  0   \n",
-       "9795          93.200          -42.0      4.076       5195.8  0   \n",
-       "9741          94.465          -41.8      4.958       5228.1  0   \n",
-       "10980         93.918          -42.7      4.963       5228.1  0   \n",
-       "15476         93.994          -36.4      4.860       5191.0  0   \n",
-       "27955         92.431          -26.9      0.740       5017.5  1   \n",
-       "7361          93.918          -42.7      4.968       5228.1  0   \n",
-       "3780          94.767          -50.8      1.044       4963.6  1   \n",
-       "15450         93.994          -36.4      4.857       5191.0  0   \n",
-       "8594          94.465          -41.8      4.967       5228.1  0   \n",
+       "16272         94.465          -41.8      4.959       5228.1  0   \n",
+       "24221         93.994          -36.4      4.856       5191.0  0   \n",
+       "15656         93.444          -36.1      4.962       5228.1  0   \n",
+       "3676          93.994          -36.4      4.857       5191.0  0   \n",
+       "16969         94.465          -41.8      4.961       5228.1  0   \n",
+       "5156          93.444          -36.1      4.968       5228.1  1   \n",
+       "4506          93.994          -36.4      4.856       5191.0  0   \n",
+       "16990         93.994          -36.4      4.857       5191.0  0   \n",
+       "6458          93.075          -47.1      1.365       5099.1  1   \n",
+       "28627         93.444          -36.1      4.965       5228.1  0   \n",
        "\n",
-       "         education_ordered      age_bins  \n",
-       "26387          high.school  (25.1, 33.2]  \n",
-       "9795     university.degree  (33.2, 41.3]  \n",
-       "9741           high.school  (25.1, 33.2]  \n",
-       "10980    university.degree  (25.1, 33.2]  \n",
-       "15476              unknown  (33.2, 41.3]  \n",
-       "27955    university.degree  (41.3, 49.4]  \n",
-       "7361     university.degree  (49.4, 57.5]  \n",
-       "3780   professional.course  (33.2, 41.3]  \n",
-       "15450          high.school  (33.2, 41.3]  \n",
-       "8594              basic.9y  (41.3, 49.4]  \n",
+       "         education_ordered        age_bins  \n",
+       "16272             basic.9y    (33.2, 41.3]  \n",
+       "24221  professional.course    (33.2, 41.3]  \n",
+       "15656    university.degree    (25.1, 33.2]  \n",
+       "3676     university.degree    (33.2, 41.3]  \n",
+       "16969          high.school    (33.2, 41.3]  \n",
+       "5156     university.degree    (25.1, 33.2]  \n",
+       "4506              basic.9y  (16.919, 25.1]  \n",
+       "16990             basic.9y    (33.2, 41.3]  \n",
+       "6458              basic.6y    (33.2, 41.3]  \n",
+       "28627          high.school    (25.1, 33.2]  \n",
        "\n",
        "[10 rows x 22 columns]"
       ]
```

Comment:
Giovanna's comments on Dmitry's work:
-  The null and missing value's from the original dataset were cleaned, the shape of the data set went from (41188, 21) to (28831, 22).
- He dropped the feature 'duration' and added 2 new columns: 'education_ordered' and 'age_bins', but the old columns 'education' and 'age' were kept. What was good for the graphic and correlation with other features, but we need to analyse if it will  make sense to keep both of each feature.
- He used WoE (Weight-of-Evidence) to treat missing values and outliers and it was a really good technique I never used before and really got my interest. This cleaning method also helped in the encoding of some features, what will is good for when we work on a  ML model.

Reply:
The quantity of rows in dataset was not reduced during cleaning. In fact I had split initial dataset on train and test parts first. And EDA is performing on train set only. So the shape of dataset has changed from (28831, 21) to (28831, 22) as no rows was dropped.
Sorry for not mentioning it at the beginning of my notebook.